### ISSUE TITLE
Usage of backslash and curly brackets in string values after migrating to react-intl v3

### ISSUE THREAD

**Issue by darkyndy** [formatjs/react-intl#1435]
### Is this a BUG REPORT or FEATURE REQUEST? (choose one)
More like a feature request, but more like a help/documentation requests.

### Expected behavior
Having a json with key/value pairs for l10n I should be able to refer to some characters without having errors from react-intl or json validator.

For example:
1.  having curly brackets in the string value as:
```
"INVALID_CHAR_CURLY_BRACES": "Curly braces (\u007B\u007D) are not allowed"
```
2. having  backslash in the string value as:
```
"INVALID_CHAR_BACK_SLASH": "Backslash (\u005C) is not allowed"
```

For the first example I shouldn't get errors from react-intl.
For both examples it should pass json validation.

### Current behavior
For the first example I get following error in console:
```
[React Intl] Error formatting message: "INVALID_CHAR_CURLY_BRACES" for locale: "en-US"
SyntaxError: Expected argNameOrNumber but "}" found.
```
For both examples it fails json validation:
```
- Expected
+ Received

  Object {
-   "INVALID_CHAR_BACK_SLASH": "ValidMessage<String>",
-   "INVALID_CHAR_CURLY_BRACES": "ValidMessage<String>",
+   "INVALID_CHAR_BACK_SLASH": "String('Backslash (\\) is not allowed') ===> Expected '#', '', 'u', '{', '}', '{', [^{}0-x1Fx7f tnr], end of input or whitespace but '' found.",
+   "INVALID_CHAR_CURLY_BRACES": "String('Curly braces ({}) are not allowed') ===> Expected '0', [1-9] or [^ tnr,.+={}#] but '}' found.",
  }  i18n-json/valid-message-syntax
```

### Step to reproduce for BUG REPORT
See link https://codesandbox.io/s/react-intl-v3-support-2l7kt where I extracted parts of the source code from my app for the purpose of demo.

Note that before migrating to v3 my l10n json for those 2 keys was:
```
{
"INVALID_CHAR_BACK_SLASH": "Backslash (\\) is not allowed",
"INVALID_CHAR_CURLY_BRACES": "Curly braces (\\{\\}) are not allowed"
}
```
where I didn't had any errors from react-intl & i18n validation.

### Your Environment

|          Executable | Version |
| ------------------: | :------ |
| `npm ls react-intl` | react-intl@3.1.8  |
|      `npm ls react` | ├── react@16.9.0 |
| `npm ls eslint-plugin-i18n-json` |  eslint-plugin-i18n-json@2.4.1 |
|     `npm --version` | 6.9.0 |
|    `node --version` | v11.9.0 |

| OS   | Version |
| ---- | ------- |
| OS X | 10.14.4 |

| Browser | Version |
| ------- | ------- |
| Chrome | 76.0.3809.100 (Official Build) (64-bit) |

**Comment by longlho** [comment]
So ICU message format syntax uses apostrophe `'` to escape things so in your example it'd be `Curly braces ('{}') are allowed`

There's no restriction on backlashes

**Comment by darkyndy** [comment]
@longlho I tested now by having:
```
"INVALID_CHAR_CURLY_BRACES": "Curly braces ('{}') are not allowed",
```
and I still get the same error from react-intl:
```
[React Intl] Error formatting message: "INVALID_CHAR_CURLY_BRACES" for locale: "en-US"
SyntaxError: Expected argNameOrNumber but "}" found.
```

And related to backslash I didn't mentioned about any restriction just that now because `'` is used as escape character it fail at json validation.

**Comment by longlho** [comment]
https://codesandbox.io/s/react-intl-v3-support-9l9jw shows no error in console for me.

https://github.com/godaddy/eslint-plugin-i18n-json is using an old version of `intl-messageformat-parser` which is why it's failing.

**Comment by darkyndy** [comment]
Thanks @longlho you are right. When I tested earlier for the curly brackets case I tested with my local app that had the strings overwritten by another source.

To conclude, having `"INVALID_CHAR_CURLY_BRACES": "Curly braces ('{}') are not allowed"` works as expected (no errors).

For the json linter thanks for looking into it, I will check their source code and come up with a PR.

Thanks again, closing this ticket.
